src/Framework/Sdk/SdkLogger.cs

```diff
@@ -15,5 +15,23 @@ public abstract class SdkLogger
         /// <param name="message">Message string.</param>
         /// <param name="messageImportance">Optional message importances. Default to low.</param>
         public abstract void LogMessage(string message, MessageImportance messageImportance = MessageImportance.Low);
+
+        /// <summary>
+        /// Logs that an event.
+        /// </summary>
+        /// <param name="args">An array of arguments to log with the event.</param>
+        public abstract void LogEvent(params object[] args);
+
+        /// <summary>
+        /// Logs that an event when an operation has started.
+        /// </summary>
+        /// <param name="args">An array of arguments to log with the event.</param>
+        public abstract void LogEventStart(params object[] args);
+
+        /// <summary>
+        /// Logs that an event when an operation has completed.
+        /// </summary>
+        /// <param name="args">An array of arguments to log with the event.</param>
+        public abstract void LogEventStop(params object[] args);
```

Comment:
I'm afraid that this is a breaking change. The class is public and actually used from outside of MSBuild. Should these methods have a default body, perhaps just delegating the call to the existing `LogEvent` in some reasonable way?

Reply:
I've made these virtual with a default implementation instead.